hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/EnhancementAsProxyLazinessInterceptor.java

```diff
@@ -39,6 +39,8 @@
 
 	private boolean initialized;
 
+	private boolean inizializing;
```

Comment:
rather than 2 booleans `initialized` and `inizializing`, should we have a single three-state field, like an enum?

That would better restrict the combinations.

Reply:
yeap, definitively  better. Done. Thanks @Sanne